DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/ToSpeckle/HostedElementConversionToSpeckle.cs

```diff
@@ -22,6 +23,22 @@ ISpeckleConverterToSpeckle converter
     _converter = converter;
   }
 
+  public void AssignHostedElements(DB.Element element, Base @base)
+  {
+    List<Base> hostedObjects = GetHostedElementsConverted(element);
+    if (hostedObjects.Count > 0)
+    {
+      if (@base.GetDetachedProp("elements") is List<Base> elements)
+      {
+        elements.AddRange(hostedObjects);
+      }
+      else
+      {
+        @base.SetDetachedProp("elements", hostedObjects);
+      }
+    }
+  }
+
```

Comment:
I'm starting to dislike this "assign pattern". it's basically a huge side-effect.

It also adds to an existing list of hosted elements, which implies we may be setting the hosted elements of a given object more than once or in parts?

This also makes it more difficult to distinguish between:
- Getting the hosted elements (Revit exclusive operation)
- Converting the hosted elements (Converter operation)
- Assigning the hosted elements (Operation done in the parent element)

Reply:
You're right that this function was doing too much. I fixed this to be more SOLID last night